src/utils.js

```diff
@@ -415,5 +421,146 @@ go.utils = {
+                var subscriptions = active_subscriptions;
+                var clean = true;  // clean tracks if api call is unnecessary
+                var patch_calls = [];
+                for (i=0; i<subscriptions.length; i++) {
+                    var updated_subscription = subscriptions[i];
+                    var endpoint = 'subscriptions/' + updated_subscription.id + '/';
+                    updated_subscription.active = false;
+                    // store the patch calls to be made
+                    patch_calls.push(function() {
+                        return go.utils.control_api_call("patch", {}, updated_subscription, endpoint, im);
+                    });
+                    clean = false;
+                }
+                if (!clean) {
+                    return Q.all(patch_calls.map(Q.try));
+            } else {
+                return Q();
+            }
+        });
+    },
+
+    switch_to_baby: function(im) {
+        var mama_id = im.user.answers.mama_id;
+        return Q
+            .all([
+                // get contact so details can be updated
+                go.utils.get_contact_by_id(mama_id, im),
+                // set existing subscriptions inactive
+                go.utils.subscriptions_unsubscribe_all(mama_id, im)
+            ])
+            .spread(function(mama_contact, unsubscribe_result) {
+                // set new mama contact details
+                mama_contact.details.baby_dob = go.utils.get_today(im.config).format('YYYY-MM-DD');
+                mama_contact.details.state_current = "baby";
+
+                // set up baby message subscription
+                baby_subscription = go.utils.setup_subscription(im, mama_contact);
+
+                return Q.all([
+                    // update mama contact
+                    go.utils.update_contact(im, mama_contact),
+                    // subscribe to baby messages
+                    go.utils.subscribe_contact(im, baby_subscription)
+                ]);
+            });
+    },
+
+    update_subscription: function(im, subscription) {
+        var endpoint = 'subscriptions/' + subscription.id + '/';
+        return go.utils
+            .control_api_call('patch', {}, subscription, endpoint, im)
+            .then(function(response) {
+                return response.data.id;
+            });
+    },
+
+    change_msg_times: function(im) {
+        var mama_id = im.user.answers.mama_id;
+        return Q
+            .all([
+                // get contact so details can be updated
+                go.utils.get_contact_by_id(mama_id, im),
+                // get existing subscriptions so schedule can be updated
+                go.utils.get_active_subscription_by_contact_id(mama_id, im)
+            ])
+            .spread(function(mama_contact, subscription) {
+                // set new mama contact details
+                mama_contact.details.voice_days = im.user.answers.state_c04_voice_days;
```

Comment:
If you're putting a function in utils I'd prefer it doesn't reach into conversation specific state reponses. Better to pass in the values. `im.user.answers.mama_id` is an exception because it's created